sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,163 @@
+/*!
+# Introduction
+
+ecs-settings-applier generates a configuration file for the ECS agent from Bottlerocket settings.
+
+*/
+
+#[macro_use]
+extern crate log;
+
+use error::SettingsApplierError;
+use serde::Serialize;
+use snafu::{OptionExt, ResultExt};
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::{env, process};
+
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+const DEFAULT_ECS_CONFIG_PATH: &str = "/etc/ecs/ecs.config.json";
+const VARIANT_ATTRIBUTE_NAME: &str = "bottlerocket.variant";
+const VERSION_ATTRIBUTE_NAME: &str = "bottlerocket.version";
+
+#[derive(Serialize, Debug, Default)]
+#[serde(rename_all = "PascalCase")]
+struct ECSConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cluster: Option<String>,
+
+    #[serde(skip_serializing_if = "std::collections::HashMap::is_empty")]
+    instance_attributes: std::collections::HashMap<String, String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    privileged_disabled: Option<bool>,
+}
+
+// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
+// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
+// https://github.com/shepmaster/snafu/issues/110
+fn main() {
+    if let Err(e) = run() {
+        eprintln!("{}", e);
+        process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
+    let args = parse_args(env::args());
+
+    // Get all settings values for config file templates
+    debug!("Requesting settings values");
+    let settings = schnauzer::get_settings(&args.socket_path).context(error::SettingsError)?;
+
+    debug!("settings = {:#?}", settings.settings);
+    let ecs = settings
+        .settings
+        .and_then(|s| s.ecs)
+        .context(error::ModelError)?;
+
+    let mut config = ECSConfig {
+        cluster: ecs.cluster,
+        privileged_disabled: ecs.allow_privileged_containers.map(|s| !s),
+        ..Default::default()
+    };
+    if let Some(os) = settings.os {
+        config
+            .instance_attributes
+            .insert(VARIANT_ATTRIBUTE_NAME.to_string(), os.variant_id);
+        config.instance_attributes.insert(
+            VERSION_ATTRIBUTE_NAME.to_string(),
+            os.version_id.to_string(),
+        );
+    }
+    if let Some(attributes) = ecs.instance_attributes {
+        for (key, value) in attributes {
+            config
+                .instance_attributes
+                .insert(key.to_string(), value.to_string());
+        }
+    }
+
+    let serialized = serde_json::to_string(&config).unwrap();
+    debug!("serialized = {}", serialized);
+
+    let config_path = PathBuf::from(DEFAULT_ECS_CONFIG_PATH);
+    write_to_disk(config_path, serialized).context(error::FSError {
```

Comment:
```suggestion
    write_to_disk(DEFAULT_ECS_CONFIG_PATH, serialized).context(error::FSError {
```
`&str` is `AsRef<Path>` :)

Reply:
...I don't know why I did that.

Reply:
Done